app/assets/javascripts/discourse/app/routes/application.js

```diff
@@ -49,6 +50,14 @@ const ApplicationRoute = DiscourseRoute.extend({
   siteSettings: service(),
   restrictedRouting: service(),
 
+  activate(transition) {
+    applyBehaviorTransformer(
+      "route-application-activate",
+      () => super.activate?.(transition),
+      { transition }
+    );
+  },
+
```

Comment:
What's this one for? Could the Ember router service events be used instead?

Reply:
The use case, I have for this one, is changing the sidebar panel before the standard sidebar is mounted to prevent a flash display before the correct panel is loaded when the page is loaded in the first time.

The admin sidebar uses the `activate` method on the route to get this result, but in the case of the admin sidebar. It has a route it can change for this effect, the docs sidebar needs to tap in existing routes.

Reply:
RouterService `on("routeDidChange", ...)` did the trick.